distributed/tests/test_steal.py

```diff
@@ -401,9 +399,6 @@ async def test_work_steal_no_kwargs(c, s, a, b):
 
 @gen_cluster(client=True, nthreads=[("127.0.0.1", 1), ("127.0.0.1", 2)])
 async def test_dont_steal_worker_restrictions(c, s, a, b):
-    future = c.submit(slowinc, 1, delay=0.10, workers=a.address)
```

Comment:
It looks like this was used to seed the task duration. I think that by removing this, this test will be vulnerable to changes to the default task duration and could potentially fail to raise.

Reply:
I see that we assert that (almost) the same scenario does indeed balance in `test_work_steal_no_kwargs`. Maybe it makes sense to adjust the delay there and add a comment that references that this tests a precondition of the others?

Reply:
I expanded on the cleanup on test_steal and, since it's unrelated to tokenization, moved it to a separate PR

Reply:
https://github.com/dask/distributed/pull/8509